module/userProjects/helper.js

```diff
@@ -1631,8 +1632,26 @@ module.exports = class UserProjectsHelper {
                         if (updatedUserProfile && updatedUserProfile.success == true && updatedUserProfile.profileMismatchFound == true) {
                             projectCreation.data.userProfile = updatedUserProfile.data;
                         }
+                        // checking the reqbody data and userLocation type matches or not
+                        for (let key in userRoleInformation) {
+                            // Skip role validation
+                            if (key === 'role') {
+                                continue;
+                            }
+                            let dataPresent;
+                            if (key === 'school') {
+                                dataPresent = projectCreation.data.userProfile.userLocations.find(location => location.code === userRoleInformation[key]);
+                            } else {
+                                dataPresent = projectCreation.data.userProfile.userLocations.find(location => location.id === userRoleInformation[key]);
+                            }
+                            if (!dataPresent) {
```

Comment:
@ankitpws the current logic, if user Profile has location A while userRoleInformation has location B then , update userProfile that will work right?